### DISCUSSION TITLE
How to enable debugging and how to set JXL_DEBUG_V_LEVEL at compile time

### DISCUSSION THREAD

**Discussion by joej970** [libjxl/libjxl#3982] [Q&A]
TLDR; 
Where and how do I need to specify the JXL_DEBUG_V_LEVEL define to enable debugging logs in the reference design?

Hello!

I am using Visual Studio 2022 for building the code. As I am using the code as a reference I need to deeply study it. I noticed there are logging statements such as (enc_frame.cc, lines 2024-2028)
 ```
JXL_DEBUG_V(2,
            "Encoding DC group #%" PRIuS " dc_y = %" PRIuS " dc_x = %" PRIuS
            " (x0, y0) = (%" PRIuS ", %" PRIuS ") (xsize, ysize) = (%" PRIuS
            ", %" PRIuS ")",
            dc_ix, dc_y, dc_x, x0, y0, xsize, ysize);
```
in enc_frame.cc file. Then in status.h there is (78-86): 
```
// JXL_DEBUG version that prints the debug message if the global verbose level
// defined at compile time by JXL_DEBUG_V_LEVEL is greater or equal than the
// passed level.
#if JXL_DEBUG_V_LEVEL > 0
#define JXL_DEBUG_V(level, format, ...) \
  JXL_DEBUG(level <= JXL_DEBUG_V_LEVEL, format, ##__VA_ARGS__)
#else
#define JXL_DEBUG_V(level, format, ...)
#endif 
```
, but I do not know how do I specify this define at compile time?

If I add "add_definitions(-DJXL_DEBUG_V_LEVEL=3)" to libjxl/CMakeLists.txt, then I get an error message: 

Error: invalid operands to binary expression ('const Kind' and 'ModularStreamId (size_t)' (aka 'jxl::ModularStreamId (unsigned long long)'))	C:\DATA\Repos\libjxl\out\build\x64-Clang-Release\libjxl	C:\DATA\Repos\libjxl\lib\jxl\dec_modular.cc	159.

The error is connected to code in dec_modular.cc, lines 155-164:
`#if JXL_DEBUG_V_LEVEL >= 1
std::string ModularStreamId::DebugString() const {
  std::ostringstream os;
  os << (kind == GlobalData   ? "ModularGlobal"
         : kind == VarDCTDC   ? "VarDCTDC"
         : kind == ModularDC  ? "ModularDC"
         : kind == ACMetadata ? "ACMeta"
         : kind == QuantTable ? "QuantTable"
         : kind == ModularAC  ? "ModularAC"`	
Also in Visual studio the linter underlines varible such as GlobalData, VarDCTDC, etc... saying "identifier GlobalData is undefined". Looks like that something gets excluded and no references are found by the compiler?
